driver/src/price_finding/naive_solver.rs

```diff
@@ -601,16 +602,26 @@ pub mod tests {
             let buy_token_price = solution.prices[order.buy_token as usize];
             let sell_token_price = solution.prices[order.sell_token as usize];
 
-            let exec_sell_amount = solution.executed_sell_amounts[i];
-            let mut exec_buy_amount = if buy_token_price > 0 {
-                (exec_sell_amount * sell_token_price).ceiled_div(buy_token_price)
+            let exec_buy_amount = solution.executed_buy_amounts[i];
+            let exec_sell_amount = if sell_token_price > 0 {
```

Comment:
New question... If we only submit buy amounts to the contract, do we really need to compute sell amounts? I know that the Snapp Contracts require this, but is that the only reason? Also, should we be looking at running the error shifting algorithm here along with the naive solver as well? I would be very excited to implement [Edmond's algorithm](https://en.wikipedia.org/wiki/Edmonds%27_algorithm) in Rust!

Reply:
I don't think the error shifting will be necessary since we only match two single orders, where one side is already the fee token. Thus we never incur any rounding errors and thus also don't need to shift any errors.

As for computing the sell amounts, this is still required for the original dƒusion system and thus should probably stay here for now.